KAFKA-19576 Fix typo in state-change log filename after rotate

jaredharley opened the pull request:
The `state-change.log` file is being incorrectly rotated to
`stage-change.log.[date]`. This change fixes the typo to have the log
file correctly rotated to `state-change.log.[date]`

_No functional changes._

Reviewers: Mickael Maison <mickael.maison@gmail.com>, Christo Lolov
 <lolovc@amazon.com>, Luke Chen <showuon@gmail.com>, Ken Huang
 <s7133700@gmail.com>, TengYao Chi <kitingiao@gmail.com>, Chia-Ping Tsai
 <chia7712@gmail.com>

jaredharley committed:
Fix typo in state-change log filename after rotate

chia7712 reviewed (approved):
LGTM

showuon commented on config/log4j2.yaml:
```diff
@@ -159,4 +159,4 @@ Configuration:
         AppenderRef:
-          ref: AuthorizerAppender
\ No newline at end of file
+          ref: AuthorizerAppender
```
Could we revert this line change?

jaredharley replied:
I've reverted this change. It got added in by Github's web editor. Thanks!

jaredharley committed:
Adding back line

jaredharley committed:
Fixing newline

jaredharley committed:
Removed final line ending

showuon commented:
LGTM! This is a nice catch! Thanks for the contribution.
But I'm afraid this change will impact the existing users after the upgrade. Let me ask in the mailing list first before merging it.

mimaison commented:
I think it's fine to merge this as this renaming was accidental and unwanted. However we should probably add a note in notable changes in 4.0.0 to indicate the name has changed.

I'm also happy to have this backported to 4.1, with also a note in the 4.1 notable changes indicating this mistake has been fixed

showuon commented:
Thanks @mimaison !
@jaredharley , could you help add a note [here](https://github.com/apache/kafka/blob/trunk/docs/upgrade.html#L54) to mention about this change?

chia7712 commented:
> notable changes in 4.0.0 to indicate the name has changed.

sounds good. @jaredharley could you please also add a note [here](https://github.com/apache/kafka/blob/trunk/docs/upgrade.html#L143) for 4.0.0?

chia7712 commented:
open https://issues.apache.org/jira/browse/KAFKA-19576 for tracing. @jaredharley please take over the jira

clolov commented:
I am happy to cut a new release candidate for 4.0.1 including this change. As Mickael mentioned, I would treat this as a bug and include it as a bug fix 😊

mimaison commented:
Ideally we'd want to merge this quickly so we can start building new RCs for 4.1.0 and 4.0.1. If @jaredharley is not able to push an update, we can merge this as is and do the cleanup in another PR.

chia7712 commented:
create https://issues.apache.org/jira/browse/KAFKA-19578 to trace the flaky

> Ideally we'd want to merge this quickly so we can start building new RCs for 4.1.0 and 4.0.1. If @jaredharley is not able to push an update, we can merge this as is and do the cleanup in another PR.

I will merge and backport this PR, and open a follow-up to update the documentation accordingly

jaredharley committed:
Added notes to upgrade guide

jaredharley commented:
Hello all, thank you for the feedback. I've just pushed the changes to the update page as requested - please let me know if you'd like to see any changes to the wording.

I am in the process of creating a Jira account so I can take ownership of [KAFKA-19576](https://issues.apache.org/jira/browse/KAFKA-19576).

chia7712 commented on docs/upgrade.html:
```diff
@@ -494,6 +497,10 @@ <h5><a id="upgrade_servers_400_notable" href="#upgrade_servers_400_notable">Nota
                     <a href="https://cwiki.apache.org/confluence/x/8ItyEg">KIP-1050</a> for more details </li>
                 </ul>
+                <li>
+                    The filename for rotated <code>state-change.log</code> files has been updated from <code>stage-change.log.[date]</code> to <code>state-change.log.[date]</code> in the log4j2.yaml configuration file.
```
@jaredharley Sorry for the unclear comment earlier, @jaredharley. Could you please create a new section, like `<h5><a id="upgrade_servers_401_notable" href="#upgrade_servers_401_notable">Notable changes in 4.0.1</a></h5>`, to mention this fix? Additionally, could you add a note to the 4.0.0 section to inform users about this bug?

jaredharley replied:
I just pushed a change to upgrade.html that I hope is what you're looking for. I used the 3.9.1 documentation as a pattern to move 4.0.0 to 4.0.1 at the top of the page, added the new 4.0.1 notable changes section, and then noted at the bottom of the 4.0.0 section that the bug exists and is corrected in 4.0.1.

Please let me know if you need any other changes!

chia7712 replied:
@jaredharley thanks for your update. Would you mind updating the line#165 accordingly?
```
check <a href="/{{version}}/documentation.html#upgrade_4_0_0">Upgrading to 4.0.0 from any version 3.3.x through 3.9.x</a> for more info.
```

jaredharley replied:
Good catch! Done and pushed.

jaredharley committed:
Clarified upgrade notes

Advanced the top-level instructions to 4.0.1, added a 4.0.1 notable changes section, and noted in 4.0.0 that the issue exists and is corrected in 4.0.1

jaredharley committed:
Correcting link

Correcting link to point at 4.0.1 anchor

showuon reviewed (approved):
LGTM!

chia7712 commented:
@mimaison I cherry-pick #19996 to 4.1 branch to eliminate conflicts

@clolov The cherry-pick to 4.0 had many conflicts, so I made some modifications locally. Pleas feel free to raise any issues if you think the doc changes aren't sufficient

chia7712 commented:
@jaredharley thanks for spotting that and provide a fix!
